cmd/dfdaemon/app/root.go

```diff
@@ -49,10 +51,28 @@ func init() {
 	opt.AddFlags(rootCmd.Flags())
 }
 
+func runTransparentProxy(opt *options.Options) error {
+	tp, err := proxy.New(g.Properties.Proxies)
+	if err != nil {
+		return fmt.Errorf("failed to create transparent proxy: %v", err)
+	}
+	s := http.Server{
+		Addr:    fmt.Sprintf(":%d", opt.ProxyPort),
+		Handler: tp,
+	}
+	logrus.Infof("launch dfdaemon transparent proxy on %s:%d", opt.HostIP, opt.ProxyPort)
+	go logrus.Fatalf("Transparent proxy stopped: %v", s.ListenAndServe())
```

Comment:
`s.ListenAndServe()` in this line will block the subsequent flows, the following codes may be what you want:

```golang
	go func() {
		logrus.Fatalf("Transparent proxy stopped: %v", s.ListenAndServe())
	}()
```

Reply:
Fixed. Some how this kind of mistakes always come back and bite me every once in a while...